Code/Framework/AzToolsFramework/Tests/Viewport/ViewBookmarkTests.cpp

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/Settings/SettingsRegistryImpl.h>
+#include <AzTest/AzTest.h>
+#include <AzToolsFramework/UnitTest/AzToolsFrameworkTestHelpers.h>
+#include <AzToolsFramework/Viewport/LocalViewBookmarkComponent.h>
+#include <AzToolsFramework/Viewport/LocalViewBookmarkLoader.h>
+
+#include <Tests/Prefab/PrefabTestFixture.h>
+
+namespace UnitTest
+{
+    struct LocalPersistentSettingsRegistry
+    {
+        AZStd::vector<char> m_buffer;
+    };
+
+    // fixture for testing the view bookmark save and load functionality
+    class ViewBookmarkTestFixture : public PrefabTestFixture
+    {
+    public:
+        void SetUpEditorFixtureImpl() override
+        {
+            PrefabTestFixture::SetUpEditorFixtureImpl();
+
+            CreateRootPrefab();
```

Comment:
Just curious, what exactly is the dependency on the root prefab and the fixture here. Are you creating one so that the 'onRootPrefabLoaded' gets triggered? I'm just wondering whether the PrefabTestFixture is really needed here or is a by-product of copying the test fixture from elsewhere. If it's not needed, maybe this test fixture doesn't have to inherit from PrefabTestFixture and solely operate like a unit test fixture.

Reply:
I think this is fine since a couple of other test fixtures create root prefab as well and they work fine. I do have some ideas to make PrefabTestFixture create a root prefab by default. I'll add you to that PR if it gets to implementation phase.

Reply:
Okay good to know, if this does change in future I can mock it out, I just wanted a way to mimic the behavior in a test as close as possible and this fixture was already doing exactly what I needed which is why I used it 🙂 Hopefully it's good enough for now and would be easy to update in future if required.